### ISSUE TITLE
Should it be "tweets.js"

### ISSUE THREAD

**Issue by edent** [timhutton/twitter-archive-parser#1]
The code currently looks for a file called `tweet.js` - but that doesn't appear in my archive.

Instead, I have tweet**s**.js and tweets-part1.js  - possibly because I post too much!

Are other archives different?

**Comment by gkatsev** [comment]
I've also got "tweets.js"

**Comment by timhutton** [comment]
Interesting!

Or it might be that the format has changed. When did you each download your archive?

I downloaded mine on 2022-04-28, with 1371 tweets (205MB), and it has `tweet.js`.

**Comment by edent** [comment]
I downloaded mine yesterday.

I think I have about 100k Tweets..........!

**Comment by timhutton** [comment]
@edent Does the code work if you change the filename? Is it just that they change the filename randomly, or have lots of other things changed too?

**Comment by edent** [comment]
Yes, it works fine.  One thing I notice is that it doesn't seem to handle retweets properly. Seems to use the old "RT:" syntax.  Have only skimmed it though.

**Comment by mozfreddyb** [comment]
I can confirm what @edent experienced. I just used your script on an export from last week. Changing the hardcoded `tweet.js` to `tweets.js` (plural) made it all work for me. Let me know if the raw data and the output.md is worth sharing anyhow.

**Comment by pepelsbey** [comment]
Same here: renamed `tweet.js` to `tweets.js` and it worked just fine with the 2022-11-06 archive.

**Comment by timhutton** [comment]
@edent The "RT:" syntax is present in the `full_text` field in the JSON they provide. Currently it just gets passed through into the markdown as-is. Do you have any thoughts on how that should be handled? Maybe just replacing 'RT:' with 'Retweet:' would be better?

**Comment by calmeilles** [comment]
According to twitter my archive was generated

> Date generated·November 11, 2022 at 5:55:38 AM GMT

and it has tweets.js   
git cloned the parser at 07:59 and it has tweet.js (no S) at line 80.  Changed that and it works. Also need to change 
`new_filename = 'data/tweets_media/'`
at line 32 otherwise references in output.md are incorrect, as below. Easily fixed after the event of course but would be nicer not to have to.

```
matthew: $ grep 1552588046175961090-FYvmfP_XgAEh6K0.jpg output.md 
@nnnnnn This morning's effort, out of the oven 3 minutes ago. ![](data/tweet_media/1552588046175961090-FYvmfP_XgAEh6K0.jpg)

matthew: $ display data/tweet_media/1552588046175961090-FYvmfP_XgAEh6K0.jpg
display-im6.q16: unable to open image `data/tweet_media/1552588046175961090-FYvmfP_XgAEh6K0.jpg': No such file or directory @ error/blob.c/OpenBlob/2874.

matthew: $ find . -name 1552588046175961090-FYvmfP_XgAEh6K0.jpg
./data/tweets_media/1552588046175961090-FYvmfP_XgAEh6K0.jpg
```

**Comment by timhutton** [comment]
@calmeilles I've added #11 for the second issue.

For the first issue:

It should cope with `tweets.js`, the code is:
```
input_filenames = glob.glob(os.path.join(data_folder, 'tweet.js')) + \
        glob.glob(os.path.join(data_folder, 'tweets.js')) + \
        glob.glob(os.path.join(data_folder, 'tweets-part*.js'))
```

Does it just parse 0 tweets or is there an error message?

**Comment by calmeilles** [comment]
Ah!  I seems to have made a mistake.  The current parser.py does indeed play nicely.

Looks like I used an earlier download in error.  Sorry for muddying the water on the first issue. (Too early, not enough coffee).
